### DISCUSSION TITLE
TypeError: unsupported operand type(s) for -: 'str' and 'str'

### DISCUSSION THREAD

**Discussion by SkiingPickle** [pvlib/pvlib-python#1945] [Q&A] [answered]
Hello, I am having trouble with the soiling function for hsu. I am checking my data types, where I have my dates as objects, and rain, pm2.5 and pm10 as floats. I keep getting a type error that says strings are trying to subtract. If anyone has any information on why this may be happening, please let me know! NONE OF MY DATA IS STR

**Comment by kandersolar** [comment]
Can you post the full error message (including traceback)?  It includes which line of code in pvlib is raising the error, which will probably give a good hint of what the problem is.

**Reply by SkiingPickle** [reply, marked as answer]
Having my dates indexed as objets was the issue, finally figured it out! Thank you!
